### PR TITLE
More uqi improvements

### PR DESCRIPTION
**Pull request by jonmagic**
## Why?

Found a few more things that could be improved based on usage at the day job.

## How?

1. Added logic to handle azure mysql connections where user must follow `<user>@<subdomain>` format.
2. Configured teardown function for each uqi client.
3. Removed engine_version property from create client function because we're not using it yet.

### CODE REVIEW

File: demo-app/src/lib/mysql-uqi-client.ts [hunk 1]
```diff
@@ -103,13 +103,22 @@ async function createMysqlUqiClient(config: MysqlConfig): Promise<UqiClient> {
     return asyncGenerator()
   }
 
+  let user = parsedUrl.username
```
Comment (hktouw) [inline demo-app/src/lib/mysql-uqi-client.ts:106]: What do you think about extracting this into a function? `extractUser(url: URL): string`?

**Review by hktouw — APPROVED** [review]
This looks great to me 🤘

File: demo-app/src/lib/mysql-uqi-client.ts [hunk 2]
```diff
@@ -136,6 +128,19 @@ async function createMysqlUqiClient(config: MysqlConfig): Promise<UqiClient> {
   })
 }
 
+function azureUserFromUri(parsedUri: URL): string {
```
Comment (hktouw) [inline demo-app/src/lib/mysql-uqi-client.ts:131]: The way this is structured isn't exclusive to azure per se, so I don't think this needs to be named `azureUserFromUri`. But not a big deal either way.

Comment (jonmagic) [reply demo-app/src/lib/mysql-uqi-client.ts:131]: The regular expression it uses is azure specific 🤷🏻‍♂️

**Pull request merged**